File: businesslib/src/main/java/com/wujia/businesslib/data/ApiResponse.java
package com.wujia.businesslib.data;

import com.wujia.lib_common.base.Constants;
import com.wujia.lib_common.data.network.BaseResponse;

public class ApiResponse<T> extends BaseResponse {
    public T data;
    public String code;
    public String msg;

    @Override
    public boolean isSuccess() {
        return Constants.HTTP_SUCESS.equals(code);
    }

    @Override
    public String getResponseCode() {
        return code;
    }

    @Override
    public String getErrorMsg() {
        return msg;
    }
}

File: businesslib/src/main/java/com/wujia/businesslib/data/ApiResponse.kt
package com.wujia.businesslib.data

import com.wujia.lib_common.base.Constants
import com.wujia.lib_common.data.network.BaseResponse

class ApiResponse<T> : BaseResponse() {
    override val isSuccess: Boolean?
        get() = Constants.HTTP_SUCESS == code
    override val responseCode: String?
        get() = code
    override val errorMsg: String?
        get() = msg
    var data: T? = null
    var code: String? = null
    var msg: String? = null

}
